### ISSUE TITLE
Can't control boiler - changing modes, updating temp works

### ISSUE THREAD

**Issue by shaarkys** [LubosD/esphome-smartboiler#5]
Hi,
I'm very much surprised after 4 years having Drazice SMART that someone was able to create this integration, well done !

I'm completely newbie with ESP32 and Home Assistant, worsened by the fact I'm running HA on Docker (no Supervised mode).
I somehow managed to run successfully your integration on DFRobot FireBeetle 2 ESP32-E and Compile it - due to some error with Git installation I had to use paths defined locally. 

I can read all the data, seems I can even adjust target temperature **but changing modes does nothing**. Is that by design or I'm missing something ? I'm checking in the meantime if I'm able to get more logs from the communication because ESP is not anymore reachable via IP but it communicates with HA normally.

Thank you.

![pic](https://user-images.githubusercontent.com/14859895/225246790-71320ed7-a35a-4b7a-91a8-2d067a35af9b.png)

```
esphome:
  name: "esphome-web-bojler"

esp32:
  board: firebeetle32
  framework:
    type: arduino

substitutions:
  mac_water_heater: RE:DA:CT:ED:51:2D

external_components:
  - source:
      type: local
      path: components
    components: [smartboiler]

# Enable Home Assistant API
api:
  encryption:
    key: REDACTED

logger:
  level: info

ble_client:
  - mac_address: ${mac_water_heater}
    id: drazice_okhe

smartboiler:
  - ble_client_id: drazice_okhe
    temp1:
      name: "Water heater temp1"
    temp2:
      name: "Water heater temp2"
    mode:
      name: "Water heater mode"
    thermostat:
      name: "Water heater thermostat"
    heat_on:
      name: "Water heater heat"
    hdo_low_tariff:
      name: "Water heater HDO"
    pin:
      name: "Water heater PIN"
    state:
      name: "Water heater state"
    version:
      name: "Water heater version"
    b_name:
      name: "Water heater name"
    consumption:
      name: "Water heater energy"

sensor:
  - platform: wifi_signal # Reports the WiFi signal strength/RSSI in dB
    name: "WiFi Signal dB"
    id: water_heater_wifi_signal_db
    update_interval: 60s
    entity_category: "diagnostic"

  - platform: copy # Reports the WiFi signal strength in %
    source_id: water_heater_wifi_signal_db
    name: "WiFi Signal Percent"
    filters:
      - lambda: return min(max(2 * (x + 100.0), 0.0), 100.0);
    unit_of_measurement: "Signal %"
    entity_category: "diagnostic"

switch:
  - platform: ble_client
    ble_client_id: drazice_okhe
    name: "Enable water heater monitoring"

ota:
  password: ""

wifi:
  ssid: REDACTED
  password: REDACTED
```

**Comment by shaarkys** [comment]
Seems this is better config for those, who would like to have also Web interface and DEBUG logs

```
esphome:
  name: "esphome-web-bojler"

esp32:
  board: firebeetle32
  framework:
    type: arduino

substitutions:
  mac_water_heater: RE:DA:CT:ED:51:2D

external_components:
  - source:
      type: local
      path: components
    components: [smartboiler]

# Enable Home Assistant API
api:
  encryption:
    key: Redacted

logger:
  level: DEBUG

# Example configuration entry
web_server:
  port: 80
  auth:
    username: Redacted
    password: Redacted

ble_client:
  - mac_address: ${mac_water_heater}
    id: drazice_okhe

smartboiler:
  - ble_client_id: drazice_okhe
    temp1:
      name: "Water heater temp1"
    temp2:
      name: "Water heater temp2"
    mode:
      name: "Water heater mode"
    thermostat:
      name: "Water heater thermostat"
    heat_on:
      name: "Water heater heat"
    hdo_low_tariff:
      name: "Water heater HDO"
    pin:
      name: "Water heater PIN"
    state:
      name: "Water heater state"
    version:
      name: "Water heater version"
    b_name:
      name: "Water heater name"
    consumption:
      name: "Water heater energy"

sensor:
  - platform: wifi_signal # Reports the WiFi signal strength/RSSI in dB
    name: "WiFi Signal dB"
    id: water_heater_wifi_signal_db
    update_interval: 60s
    entity_category: "diagnostic"

  - platform: copy # Reports the WiFi signal strength in %
    source_id: water_heater_wifi_signal_db
    name: "WiFi Signal Percent"
    filters:
      - lambda: return min(max(2 * (x + 100.0), 0.0), 100.0);
    unit_of_measurement: "Signal %"
    entity_category: "diagnostic"

switch:
  - platform: ble_client
    ble_client_id: drazice_okhe
    name: "Enable water heater monitoring"

ota:
  safe_mode: true
  password: Redacted

wifi:
  ssid: APRedacted
  password: Redacted

```

**Comment by shaarkys** [comment]
So now I'm able also to provide some logs ... at this moment boiler heats and trying to stop it (as example)... heater mode STOP

```
10:29:39	[D]	[select:062]	
'Water heater mode' - Setting
10:29:39	[D]	[select:115]	
'Water heater mode' - Set selected option to: STOP
10:29:39	[D]	[smartboiler:067]	
Sending: REQ: 27 DATA=[1B.00.01.00.00.00.00.00 (8)]
10:29:39	[D]	[smartboiler:425]	
Queue status - QUEUE=[0], SENT_QUEUE=[1]
10:29:39	[D]	[smartboiler:219]	
Received: REQ: 51 DATA=[35.31.01.00.00.00.00.00.00.00.00.00.00.00.00.00.00.00.00.00 (20)]
10:29:39	[D]	[smartboiler:232]	
Received confirmation for packet with UID: 01
10:29:39	[D]	[smartboiler:233]	
mByteData: DATA=[00.00.00.00.00.00.00.00.00.00.00.00.00.00.00.00 (16)]
10:29:39	[D]	[smartboiler:239]	
original request was: 27
10:29:39	[D]	[smartboiler:239]	
original request was: 27
10:29:39	[D]	[smartboiler:219]	
Received: REQ: 20 DATA=[32.30.30.30.30.00.00.00.00.00.00.00.00.00.00.00.00.00.00.00 (20)]
10:29:39	[D]	[smartboiler:067]	
Sending: REQ: 18 DATA=[12.00.02.00.00.00.00.00 (8)]
10:29:39	[D]	[smartboiler:425]	
Queue status - QUEUE=[0], SENT_QUEUE=[1]
10:29:39	[D]	[smartboiler:219]	
Received: REQ: 52 DATA=[35.32.30.30.30.30.32.00.00.00.00.00.00.00.00.00.00.00.00.00 (20)]
10:29:43	[D]	[smartboiler:219]	
Received: REQ: 11 DATA=[31.31.32.2E.31.30.3A.33.30.3A.30.35.00.00.00.00.00.00.00.00 (20)]
```

Change of MODE to OFF (from HEAT)

```
10:31:08	[D]	[climate:014]	
  Mode: OFF
10:31:13	[D]	[smartboiler:219]	
Received: REQ: 11 DATA=[31.31.32.2E.31.30.3A.33.31.3A.33.35.00.00.00.00.00.00.00.00 (20)]
10:31:18	[D]	[smartboiler:219]	
Received: REQ: 11 DATA=[31.31.32.2E.31.30.3A.33.31.3A.34.30.00.00.00.00.00.00.00.00 (20)]
10:31:20	[D]	[esp32_ble_tracker:238]	
Starting scan...
10:31:23	[D]	[smartboiler:219]	
Received: REQ: 11 DATA=[31.31.32.2E.31.30.3A.33.31.3A.34.35.00.00.00.00.00.00.00.00 (20)]
10:31:28	[D]	[smartboiler:219]	
Received: REQ: 11 DATA=[31.31.32.2E.31.30.3A.33.31.3A.35.30.00.00.00.00.00.00.00.00 (20)]
10:31:28	[D]	[climate:010]	
'Water heater thermostat' - Setting
10:31:28	[D]	[climate:014]	
  Mode: HEAT
10:31:33	[D]	[smartboiler:219]	
Received: REQ: 11 DATA=[31.31.32.2E.31.30.3A.33.31.3A.35.35.00.00.00.00.00.00.00.00 (20)]
10:31:38	[D]	[smartboiler:219]	
Received: REQ: 11 DATA=[31.31.32.2E.31.30.3A.33.32.3A.30.30.00.00.00.00.00.00.00.00 (20)]
```

**Comment by pedobry** [comment]
It's a bug in the code. I created pull request with a fix #7

**Comment by LubosD** [comment]
Thanks!
